extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java

```diff
@@ -877,6 +917,17 @@ public Builder setTableDefinition(final TableDefinition tableDefinition) {
             return this;
         }
 
+        /**
+         * Arrays containing the column names for indexes to persist. The write operation will store the index info as
+         * sidecar tables. This argument is used to narrow the set of indexes to write, or to be explicit about the
+         * expected set of indexes present on all sources. Indexes that are specified but missing will be computed on
+         * demand.
+         */
+        public Builder setIndexColumns(final String[][] indexColumns) {
```

Comment:
`addIndexColumns` might be a better pattern for end-users.
Have two overloads. One takes an `Iterable<String[]>`, other takes a `String...`. Could allow `Iterable<Collection<String>>`, but not really necessary.

If you're switching to `Collection` internally, then you might also want `addIndexColumns(Collection<String>)`